### PR TITLE
Check for error messages in url

### PR DESCRIPTION
**Pull request by Niharika29**
CC @kaldari

### CODE REVIEW

File: IABot/checkIfDead.php [hunk 1]
```diff
@@ -132,6 +132,12 @@ protected function processResult( $headers, $curlerrno, $url ) {
 			} else {
 				return true;
 			}
+			// Check for error messages in redirected URL string
+		} elseif ( strpos( $effectiveUrlClean, '404.htm' ) !== false ||
+				   strpos( $effectiveUrlClean, '/404/' ) !== false ||
+				   stripos( $effectiveUrlClean, 'error' ) !== false ||
```
Comment (ricordisamoa) [inline IABot/checkIfDead.php]: Wouldn't this line mark URLs like http://expressjs.com/en/guide/error-handling.html as dead?

Comment (Niharika29) [inline IABot/checkIfDead.php]: You're right. I removed this in a follow-up patch. Thank you!

**Comment by wumpus** [comment]
Looking for /404/ or /error/ in redirect urls is a good way to discover candidate "soft 404" pages, and then you can confirm it by fetching a couple of randomly-generated urls which ought to 404.

**Comment by Niharika29** [comment]
> Looking for /404/ or /error/ in redirect urls is a good way to discover candidate "soft 404" pages, and then you can confirm it by fetching a couple of randomly-generated urls which ought to 404.
 
Thanks for your input, @wumpus.

**Pull request merged**